test/server.py

```diff
@@ -42,12 +42,14 @@ def __init__(self,
                  host='localhost',
                  ready_event=None,
                  h2=True,
-                 secure=True):
+                 secure=True,
+                 secure_auto_wrap_socket=True):
```

Comment:
It'd be nicer, rather than adding boolean fields that only make sense in the presence of other boolean fields, to turn `secure` into a field that accepts an `enum`, and have that `enum` include the relevant options.